app/controllers/pieces_controller.rb

```diff
@@ -19,7 +19,7 @@ def update
   private
   
     def set_piece
-      @piece = Piece.find(params[:id])
+      @piece ||= Piece.find(params[:id])
```

Comment:
I was aiming for memoization with using `set_piece` and the `||=` technique, not to mention a minor performance savings in the event that a player _picked up_ a piece but didn't actually move it (since dropping the piece triggers the update action). This wouldn't forfeit their turn, and if said player decided to actually move the piece ultimately, that object would already be in the sever's memory and not require another trip to the database. At least I think that's what I'm accomplishing here.